HaveValidSnapshot.swift

```diff
@@ -67,8 +72,15 @@ func _getDefaultReferenceDirectory(sourceFileName: String) -> String {
 
     let pathComponents: NSArray = (sourceFileName as NSString).pathComponents
     for folder in pathComponents {
+        var found = false
```

Comment:
Cool functionality, I feel we could be a little more clever about it. What about:

```swift
let found = testFolderSuffixes.filter { $0.lowerCaseString.hasSuffix(suffix.lowerCaseString) }.count > 0
```

Does that make sense?